packages/nx/src/nx-cloud/update-manager.ts

```diff
@@ -0,0 +1,364 @@
+    )
+  ) {
+    const axios = createApiAxiosInstance(options);
+
+    let verifyBundleResponse: AxiosResponse<VerifyClientBundleResponse>;
+    try {
+      verifyBundleResponse = await verifyCurrentBundle(axios, currentBundle);
+    } catch (e: any) {
+      // Enterprise image compatibility, to be removed
+      if (e.message === 'Request failed with status code 404' && options.url) {
+        console.warn(
+          '[WARNING] Nx Cloud was unable to fetch or verify its bundle from the instance hosted at: ',
+          options.url
+        );
+        console.warn(
+          '[WARNING] Nx Cloud will continue to function as expected, but your installation should be updated'
+        );
+        console.warn('[WARNING] soon for the best experience.');
+        console.warn('[WARNING] ');
+        console.warn(
+          '[WARNING] If you are an Nx Enterprise customer, please reach out to your assigned Developer Productivity Engineer.'
+        );
+        console.warn('[WARNING] ');
+        console.warn(
+          '[WARNING] If you are NOT an Nx Enterprise customer but are seeing this message, please reach out to cloud-support@nrwl.io.'
+        );
+        console.warn('[WARNING] ');
+        console.warn(
+          '[WARNING] To prevent this check on startup, set `useLightClient: false` in your task runner options found in `nx.json`.'
+        );
+
+        return {
+          version: 'NX_ENTERPRISE_OUTDATED_IMAGE',
+          fullPath: '',
+        };
+      }
+
+      debugLog(
+        'Could not verify bundle. Resetting validation timer and using previously installed or default runner. Error: ',
+        e
+      );
+      writeBundleVerificationLock(runnerBundleInstallDirectory);
+
+      return currentBundle;
+    }
+
+    if (verifyBundleResponse.data.valid) {
+      debugLog('Currently installed bundle is valid');
+      writeBundleVerificationLock(runnerBundleInstallDirectory);
+      return currentBundle!!;
+    }
+
+    const { version, url } = verifyBundleResponse.data;
+    debugLog(
+      'Currently installed bundle is invalid, downloading version',
+      version,
+      ' from ',
+      url
+    );
+    return {
+      version,
+      fullPath: await downloadAndExtractClientBundle(
+        axios,
+        runnerBundleInstallDirectory,
+        version,
+        url
+      ),
+    };
+  }
+
+  return currentBundle!!;
+}
+
+function findWorkspaceRoot(startPath) {
+  let currentPath = isAbsolute(startPath) ? startPath : resolve(startPath);
+
+  while (currentPath !== dirname(currentPath)) {
+    const potentialFile = join(currentPath, 'nx.json');
+    if (existsSync(potentialFile)) {
+      return currentPath;
+    }
+    currentPath = dirname(currentPath);
+  }
+
+  return null;
+}
+
+function getRunnerBundleInstallDirectory(
+  options: CloudTaskRunnerOptions
+): string {
+  const cacheDirectory =
+    (process.env.NX_CACHE_DIRECTORY || options.cacheDirectory) ??
+    join('node_modules', '.cache', 'nx');
```

Comment:
Is the default cache dir defined anywhere in the `nx` package? Otherwise should be updated to the `.nx/cache` location